packages/world/src/AccessControl.sol

```diff
@@ -44,7 +44,7 @@ library AccessControl {
   ) internal view returns (bytes32 resourceSelector) {
     resourceSelector = ResourceSelector.from(namespace, file);
 
-    if (NamespaceOwner.get(namespace) != msg.sender) {
+    if (NamespaceOwner.get(namespace) != caller) {
```

Comment:
this seems like a scary thing to have missed! do we have tests for this?

Reply:
Added some!